Apps/WebClient/src/Server/Controllers/NoteController.cs

```diff
@@ -0,0 +1,121 @@
+            object noteService,
+            IHttpContextAccessor httpContextAccessor,
+            IAuthorizationService authorizationService)
+        {
+            this.noteService = noteService;
+            this.httpContextAccessor = httpContextAccessor;
+            this.authorizationService = authorizationService;
+        }
+
+        /// <summary>
+        /// Posts a patient note json to be inserted into the database.
+        /// </summary>
+        /// <returns>The http status.</returns>
+        /// <param name="hdid">The patient hdid.</param>
+        /// <param name="model">The patient note request model.</param>
+        /// <response code="200">The note record was saved.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpPost]
+        [Route("{hdid}")]
+        [Authorize(Policy = "PatientOnly")]
+        public async Task<IActionResult> CreateNote(string hdid, [FromBody] Database.Models.Note model)
+        {
+            // Validate that the query parameter matches the post body
+            if (string.IsNullOrEmpty(hdid) ||
+                model == null ||
+                !hdid.Equals(model!.HdId, StringComparison.CurrentCultureIgnoreCase))
+            {
+                return new BadRequestResult();
+            }
+
+            // Validate the hdid to be a patient.
+            ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
+            AuthorizationResult isAuthorized = await this.authorizationService
+                .AuthorizeAsync(user, hdid, PolicyNameConstants.UserIsPatient)
+                .ConfigureAwait(true);
+            if (!isAuthorized.Succeeded)
+            {
+                return new ForbidResult();
+            }
+
+            //RequestResult<Database.Models.Note> result = this.noteService.CreateNote(model);
+            return new JsonResult(new object());
+        }
+
+        /// <summary>
+        /// Gets all notes for the specified user.
+        /// </summary>
+        /// <returns>The list of notes model wrapped in a request result.</returns>
+        /// <param name="hdid">The user hdid.</param>
+        /// <response code="200">Returns the list of notes.</response>
+        /// <response code="401">the client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpGet]
+        [Route("{hdid}")]
+        public async Task<IActionResult> GetAll(string hdid)
```

Comment:
To make it restful, lest make the route /Note/ and /Note/{noteId} if necessary to retrieve a single note